### PR TITLE
fix: Forward auth token

### PR DESCRIPTION
**Pull request by deekshas8**
Closes SAP/cloud-sdk-backlog#1084.

- `userJWT` should be passed in authorization header of the request when `forwardAuthToken` is enabled, irrespective of the authentication type.
- Refactored `getMergedHeaders`

### CODE REVIEW

**Review by tomfrenken — APPROVED** [review]
LGTM, just one comment regarding the test

File: packages/connectivity/src/scp-cf/authorization-header.spec.ts [hunk 1]
```diff
@@ -59,14 +59,22 @@ describe('buildAuthorizationHeaders', () => {
       ).resolves.not.toThrow();
     });
 
-    it('creates no authentication headers when only url is defined in a destination.', async () => {
+    it('does not create authentication headers when only url is defined in a destination.', async () => {
```
[the remark this answers is not in the document]
Comment (deekshas8) [reply packages/connectivity/src/scp-cf/authorization-header.spec.ts]: Done.

**Review by marikaner — APPROVED** [review]
LGTM, but I left a few comments.

File: packages/connectivity/src/scp-cf/authorization-header.spec.ts [hunk 2]
```diff
@@ -53,20 +53,23 @@ describe('buildAuthorizationHeaders', () => {
       ).resolves.not.toThrow();
     });
 
-    it('defaults to NoAuthentication', async () => {
-      await expect(
-        buildAuthorizationHeaders({ url: 'https://example.com' })
-      ).resolves.not.toThrow();
+    it('defaults to NoAuthentication and does not create authentication headers when only url is defined', async () => {
```
[the remark this answers is not in the document]
Comment (deekshas8) [reply packages/connectivity/src/scp-cf/authorization-header.spec.ts:57]: Added a check to test for authentication type.

**Pull request merged**